build_tools/ci/cpu_comparison/run.py

```diff
@@ -458,6 +460,9 @@ def generate_aie_vmfb(
 
     if collapse_unit_dims:
         compilation_flags += ["--iree-amdaie-enable-collapsing-unit-dims"]
+    
+    if function_outline:
+        compilation_flags += ["--iree-amdaie-enable-function-outlining"]
```

Comment:
It has to be two separate flags?

Reply:
My opinion is that it makes sense to have separate flags, and I don't see any harm in separate flags. @Abhishek-Varma maybe just add a comment with each of the flags that it "if enabled, it might reduce program memory usage"?

Reply:
> but you have to be very familiar with the project in order to use all correct flags for compiling.

The defaults in this case are fine. My perspective is that these flags are just for the developer to play with and make progress without forcing them on all pipelines. Eventually, some of them will be expired. Coalescing for example: once the upstream changes are done and integrated (the pass to coalesce and collapse dimensions simultaneously which I think Mahesh mentioned) we can always enable coalescing, and the flag can be removed. 

If we had a single flag what would it be called 
(1)  `--iree-amdaie-optimize-for-program-memory` or 
(2) `--iree-amdaie-enable-outlining-and-dim-collapsing-and-coalescing`` ? 

I suppose we could go with (1) but it seems like we don't understand what these changes do well enough yet to give it such a bold name. 

Option (2) seems strange, why not just have them as separate flags.

LLVM has at 9 flags just to control loop unrolling https://llvm.org/doxygen/LoopUnrollPass_8h_source.html (!!)

Reply:
I'm totally okay with separate flags for experimenting. Just saying it'd be good to keep less flags for production. @Abhishek-Varma could you add more comments on directions of when this flag `--iree-amdaie-enable-collapsing-unit-dims` is needed?

Reply:
I've added it in https://github.com/nod-ai/iree-amd-aie/pull/896.